packages/core/src/modules/connections/DidExchangeProtocol.ts

```diff
@@ -523,64 +523,73 @@ export class DidExchangeProtocol {
+          const signedDid = TypedArrayEncoder.fromBase64(base64UrlPayload).toString()
 
-      if (signedDid !== message.did) {
-        throw new CredoError(
-          `DID Rotate attachment's did ${message.did} does not correspond to message did ${message.did}`
-        )
-      }
-
-      const { isValid, signerKeys } = await this.jwsService.verifyJws(agentContext, {
-        jws: {
-          ...jws,
-          payload: base64UrlPayload,
-        },
-        jwkResolver: ({ jws: { header } }) => {
-          if (typeof header.kid !== 'string' || !isDid(header.kid, 'key')) {
-            throw new CredoError('JWS header kid must be a did:key DID.')
+          if (signedDid !== message.did) {
+            throw new CredoError(
+              `DID Rotate attachment's did ${message.did} does not correspond to message did ${message.did}`
+            )
           }
 
-          const didKey = DidKey.fromDid(header.kid)
-          return getJwkFromKey(didKey.key)
-        },
-      })
-
-      if (!isValid || !signerKeys.every((key) => invitationKeysBase58?.includes(key.publicKeyBase58))) {
-        throw new DidExchangeProblemReportError(
-          `DID Rotate signature is invalid. isValid: ${isValid} signerKeys: ${JSON.stringify(
-            signerKeys
-          )} invitationKeys:${JSON.stringify(invitationKeysBase58)}`,
-          {
-            problemCode: DidExchangeProblemReportReason.ResponseNotAccepted,
+          const { isValid, signerKeys } = await this.jwsService.verifyJws(agentContext, {
+            jws: {
+              ...jws,
+              payload: base64UrlPayload,
+            },
+            jwkResolver: ({ jws: { header } }) => {
+              if (typeof header.kid !== 'string' || !isDid(header.kid, 'key')) {
+                throw new CredoError('JWS header kid must be a did:key DID.')
+              }
+
+              const didKey = DidKey.fromDid(header.kid)
+              return getJwkFromKey(didKey.key)
+            },
+          })
+
+          if (!isValid || !signerKeys.every((key) => invitationKeysBase58?.includes(key.publicKeyBase58))) {
+            throw new DidExchangeProblemReportError(
+              `DID Rotate signature is invalid. isValid: ${isValid} signerKeys: ${JSON.stringify(
+                signerKeys
+              )} invitationKeys:${JSON.stringify(invitationKeysBase58)}`,
+              {
+                problemCode: DidExchangeProblemReportReason.ResponseNotAccepted,
+              }
+            )
           }
-        )
+        } catch (e) {
+          this.logger.warn(`Document does not contain didRotate. Error: ${e.message}`)
+        }
       }
     }
 
     // Now resolve the document related to the did (which can be either a public did or an inline did)
     try {
-      return await agentContext.dependencyManager.resolve(DidsApi).resolveDidDocument(message.did)
+      if (message.did.startsWith('did:'))
```

Comment:
In which cases the `did` field in a DID Exchange Request or Response would not start by `did:`?

Reply:
This came up as an issue with testing against aca-py and being sent an unqualified did in response to using `did:peer:1`. Aca-py supports processing `did:peer:1` but will not send one back which is strange behavior on their part. I would suggest taking this change for now and opening an issue with Aca-py to try and point out this behavior. I might not be right but I was under the impression that unqualified dids were going to the way side here soon.

Reply:
Well, I'd rather suggest to fix the issue in ACA-Py so both are working as expected :stuck_out_tongue: . Jokes apart, we are not supporting unqualified DIDs in DID Exchange protocol so I think it is better to keep it like that.